mailbox/api/src/main/java/org/apache/james/mailbox/AttachmentManager.java

```diff
@@ -33,5 +33,7 @@
 
     List<Attachment> getAttachments(List<AttachmentId> attachmentIds, MailboxSession mailboxSession) throws MailboxException;
 
-    void storeAttachment(Attachment attachment, MailboxSession mailboxSession) throws MailboxException;
+    void storeAttachment(Attachment.WithBytes attachment, MailboxSession mailboxSession) throws MailboxException;
+
+    Attachment.WithBytes retrieveContent(AttachmentId id, MailboxSession session) throws MailboxException, AttachmentNotFoundException;
```

Comment:
The fact that Attachment.WithBytes contains almost the same thing as Attachment did makes its use more general than it could and the lifetime of the byte array much longer than needed, too.
What about returning an InputStream here? Did you tried? Is it too disruptive in the codebase?

Reply:
I share your concern but:

 - 1. this is what I came up to given a limited working time on this topic, which end up not being required by my current contribution objectives (Which are: OpenPaaS INBOX GetMessages requested performed upon mail listing should not result on Body reads in the blobStore). I allocated a half day of my working time to make progress on that important topic. But that end up being merely related to my overall goals.

I'd rather take this step as a valid state toward a better approach over attachment handling.

 - 2. Well, we need a repeatable input stream. And yes that is disruptive on the write path.

Let me have a look for the read path.

Reply:
That's contributed.